### PR TITLE
Feat/1787 site map

### LINKED ISSUE #1787: [Dev-ready Design]
**Description:**
Provide grantees an alternative route to access TDP links in the form of a site map. 

**Related User Stories:**

—As a [Grantee] I want a site map page so that I can easily access TDP links to help me get to where I need to go next. 

**Recommended pa11y actions:**

—‘click element #error-report’,—

**AC:**

- [ ] Documentation work for the following has occurred:
   - [ ] Relevant User stories.
   - [ ] Recommended pa11y checks.
   - [ ] Updating living UX documents, e.g. User Flows or Personas(if relevant).
- [ ] Dev/Design sync has occurred; resulting tickets created
- [ ] The design is usable and accessible, meaning it adheres to definition of done standards for design work.
> <details>
>      <summary>Expand Definition of Done Standards</summary>
> 
> - It uses [USWDS components and follows it’s UX guidance](https://designsystem.digital.gov/components/), or a deviation is clearly documented
> 
> - Language is intentional and [plain](https://plainlanguage.gov/guidelines/); placeholders are clearly documented
> - It follows [accessibility guidelines](https://accessibility.digital.gov/) and accessibility implementation notes are documented (e.g. clear information hierarchy, color is not the only way meaning is communicated, etc.)
> - If feedback identifies bigger questions or unknowns, create additional issues to investigate
> </details>

**Tasks:**

- [ ] —Draft design in Figma—

**Supporting Documentation:**

—Link to Figma and Dev-Ready Design folder PDF—

### PR DESCRIPTION
**Pull request by riatzukiza**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1787 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```
1. Check that there is no site map link on the spash page
2. Check that you can't directly navigate to http://localhost:3000/site-map
1. sign in.
4. click site map 
5. try all the links make sure they work
6. 

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Comment by riatzukiza** [comment]
I need to write tests for this, otherwise it is ready for code/a11y reviews.

File: tdrs-frontend/src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -20,6 +20,16 @@ function Footer() {
                     Privacy policy
                   </a>
                 </li>
+                <li className="mobile-lg:grid-col-6 desktop:grid-col-auto usa-footer__primary-content">
+                  <a
+                    className="usa-footer__primary-link"
+                    href="/site-map"
+                    target="_blank"
+                    rel="noreferrer"
```
Comment (n0remac) [inline tdrs-frontend/src/components/Footer/Footer.jsx:28]: add noopener

File: tdrs-frontend/src/components/SiteMap/SiteMap.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  <div className="margin-top-5">
+    <SiteMap.Link text="Home" link="/home" />
+    <SiteMap.Link
+      text="Privacy Policy"
+      link="https://www.acf.hhs.gov/privacy-policy"
+    />
+    <SiteMap.Link text="Data Files" link="/data-files" />
+    <SiteMap.Link text="Profile" link="/profile" />
+    <SiteMap.Link
+      text="Home"
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/SiteMap/SiteMap.jsx:13]:
```suggestion
      text="Admin"
```

File: tdrs-frontend/src/components/Footer/Footer.jsx [hunk 3]
```diff
@@ -20,6 +22,18 @@ function Footer() {
                     Privacy policy
                   </a>
                 </li>
+                {authenticated ? (
+                  <li className="mobile-lg:grid-col-6 desktop:grid-col-auto usa-footer__primary-content">
+                    <a
+                      className="usa-footer__primary-link"
+                      href="/site-map"
+                      target="_blank"
+                      rel="noreferrer"
```
Comment (n0remac) [inline tdrs-frontend/src/components/Footer/Footer.jsx:31]: add noopener

File: tdrs-frontend/src/components/Footer/Footer.jsx [hunk 4]
```diff
@@ -20,6 +22,18 @@ function Footer() {
                     Privacy policy
                   </a>
                 </li>
+                {authenticated ? (
+                  <li className="mobile-lg:grid-col-6 desktop:grid-col-auto usa-footer__primary-content">
```
Comment (n0remac) [inline tdrs-frontend/src/components/Footer/Footer.jsx:26]: I never used mobile-lg. We should probably look through the frontend together at some point to get our styling as consistent as possible. Maybe we can save something like that for when we don't have such a big deadline.

File: tdrs-frontend/src/components/SiteMap/SiteMap.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React, { useRef, useState } from 'react'
+
+const SiteMap = () => (
+  <div className="margin-top-5">
+    <SiteMap.Link text="Home" link="/home" />
```
Comment (n0remac) [inline tdrs-frontend/src/components/SiteMap/SiteMap.jsx:5]: Can you explain what is going on here? You are referencing the SiteMap variable as it's building itself?

File: tdrs-frontend/src/components/SiteMap/SiteMap.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import React, { useRef, useState } from 'react'
+
+const SiteMap = () => (
```
Comment (n0remac) [inline tdrs-frontend/src/components/SiteMap/SiteMap.jsx:3]: Maybe, just add a note here to remind people to update the tests when a new link is added

**Review by n0remac — CHANGES_REQUESTED** [review]
Only a couple little changes. All the code looks really good.

File: tdrs-frontend/src/components/Footer/Footer.jsx (hunk 4 above)
Comment (n0remac) [reply tdrs-frontend/src/components/Footer/Footer.jsx:26]: I changed the footer in my pr too 
https://github.com/raft-tech/TANF-app/pull/1924/files#diff-9fabec8f76eaeefca82056f5b7fb1288c54eaaa656dcfcb54473eb4413102796

File: tdrs-frontend/src/components/SiteMap/SiteMap.test.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      user: {
+        email: 'test@test.com',
+        roles: [{ id: 1, name: 'Developer', permissions: [] }],
+        access_request: true,
+      },
+      authenticated: true,
+    },
+  }
+
+  it('When an authenticated Developer visits the sitemap', () => {
```
Comment (n0remac) [inline tdrs-frontend/src/components/SiteMap/SiteMap.test.js:21]: Developer should see admin

Comment (n0remac) [reply tdrs-frontend/src/components/SiteMap/SiteMap.test.js:21]: test all roles

**Review by n0remac — APPROVED** [review]

**Pull request closed without merging**